src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -134,8 +134,11 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         dispatcher.RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [weakThis{ get_weak() }]() {
             if (auto strongThis{ weakThis.get() })
             {
-                // The event that is raised when the text selection is modified.
-                strongThis->RaiseAutomationEvent(AutomationEvents::TextPatternOnTextSelectionChanged);
+                if (auto control{ strongThis->_termControl.get() })

```

Comment:
Tbh from the PR description alone I haven't understood why we need to hold a strong reference to `control` across the `RaiseAutomationEvent` call. What if `RaiseAutomationEvent` is doing anything asynchronously? In that case holding `control` wouldn't fix anything... 🤔

Reply:
Honestly, I'm not entirely sure this'll fix it. Without a solid repro, it was hard for me to figure out how exactly we were getting into this place where a com_ptr was being released, and what exactly the pointer was supposed to be pointing at. This is definitely a "maybe it works, if it doesn't, let's revert and try something else" kinda crash fix.

Reply:
Regardless, passing in a raw pointer to an impl type has almost certainly got to be wrong.